src/lib/y2network/autoinst_profile/routing_section.rb

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/autoinst_profile/section_with_attributes"
+require "y2network/autoinst_profile/route_section"
+
+module Y2Network
+  module AutoinstProfile
+    # This class represents an AutoYaST <routing> section under <networking>
+    #
+    #  <routing>
+    #    <ipv4_forward config:type="boolean">false</ipv4_forward>
+    #    <ipv6_forward config:type="boolean">false</ipv6_forward>
+    #    <routes config:type="list">
+    #      <route> <!-- see RouteSection class -->
+    #        <destination>192.168.1.0</destination>
+    #        <device>eth0</device>
+    #        <extrapara>foo</extrapara>
+    #        <gateway>-</gateway>
+    #        <netmask>-</netmask>
+    #      </route>
+    #    </routes>
+    #  </routing>
+    #
+    # @see NetworkingSection
+    class RoutingSection < SectionWithAttributes
+      def self.attributes
+        [
+          { name: :ipv4_forward },
+          { name: :ipv6_forward },
+          { name: :routes }
+        ]
+      end
+
+      define_attr_accessors
+
+      # @!attribute ipv4_forward
+      #  @return [Boolean]
+
+      # @!attribute ipv6_forward
+      #  @return [Boolean]
+
+      # @!attribute routes
+      #   @return [Array<RouteSection>]
+
+      # Clones network routing settings into an AutoYaST routing section
+      #
+      # @param routing [Y2Network::Routing] Routing settings
+      # @return [RoutingSection]
+      def self.new_from_network(routing)
+        result = new
+        initialized = result.init_from_network(routing)
+        initialized ? result : nil
+      end
+
+      # Constructor
+      def initialize(_parent = nil)
```

Comment:
just note when you just need to pass all arguments to parent, you call use simple `*args` and pass it. So in future you do not need to modify this method if parent need to change arguments.